db/00000000000001_initial_schema.up.sql

```diff
@@ -226,11 +226,23 @@ CREATE TABLE report_aggregations(
     client_report_id    BYTEA NOT NULL,                     -- the client report ID this report aggregation is associated with
     client_timestamp    TIMESTAMP NOT NULL,                 -- the client timestamp this report aggregation is associated with
     ord                 BIGINT NOT NULL,                    -- a value used to specify the ordering of client reports in the aggregation job
-    state               REPORT_AGGREGATION_STATE NOT NULL,  -- the current state of this report aggregation
-    helper_prep_state   BYTEA,                              -- the current VDAF prepare state (opaque VDAF message, only if in state WAITING, only populated for helper)
-    leader_prep_transition BYTEA,                           -- the current VDAF prepare transition (opaque VDAF message, only if in state WAITING, only populated for leader)
-    error_code          SMALLINT,                           -- error code corresponding to a DAP ReportShareError value; null if in a state other than FAILED
     last_prep_resp      BYTEA,                              -- the last PrepareResp message sent to the Leader, to assist in replay (opaque DAP message, populated for Helper only)
+    state               REPORT_AGGREGATION_STATE NOT NULL,  -- the current state of this report aggregation
+
+    -- Additional data for state StartLeader.
+    public_share                  BYTEA,  -- the public share for the report (opaque VDAF message)
+    leader_extensions             BYTEA,  -- encoded sequence of Extension messages from Leader input share (opaque DAP messages)
+    leader_input_share            BYTEA,  -- encoded leader input share (opaque VDAF message)
+    helper_encrypted_input_share  BYTEA,  -- encoded HPKE ciphertext of helper input share (opaque DAP message)
```

Comment:
WDYT about writing some `CONSTRAINT`s to ensure that the columns are non-NULL or NULL as expected given the value of `state`, akin to the ones on table `tasks` that enforce teh presence or absence of auth tokens?

Reply:
I'd like to keep as much functionality in our application servers (which are horizontally scalable) rather than our DB (which is a monolith which must scale vertically). That is, I prefer to keep these checks in our datastore code rather than encoding them in our DB directly, especially for something like `report_aggregations` which constitutes a large fraction of our total writes.

That said, this is based on ~vibes~; if you feel strongly I can change things.

Reply:
I don't have a particularly strong argument for doing it in the database vs. code.